crates/hir_ty/src/infer/expr.rs

```diff
@@ -21,15 +21,19 @@ use crate::{
     primitive::{self, UintTy},
     traits::{FnTrait, InEnvironment},
     utils::{generics, variant_data, Generics},
-    Binders, CallableDefId, FnPointer, FnSig, Obligation, OpaqueTyId, Rawness, Scalar, Substs,
-    TraitRef, Ty,
+    AdtId, Binders, CallableDefId, FnPointer, FnSig, Obligation, OpaqueTyId, Rawness, Scalar,
+    Substs, TraitRef, Ty,
 };
 
 use super::{
     find_breakable, BindingMode, BreakableContext, Diverges, Expectation, InferenceContext,
     InferenceDiagnostic, TypeMismatch,
 };
 
+fn adt_ty(adt: hir_def::AdtId, substs: Substs) -> Ty {
```

Comment:
could we add that as an inherent method to `Ty`?

Reply:
Was thinking of how this will be a foreign type in the future, but I guess we will want some extension traits at some point anyways.

Reply:
Yeah, I assume we'll want to refactor that anyway. I think we'll need to see what other methods we need there.

Reply:
Ye most likely.

On another Note, I also looked a bit into moving TraitId to chalk and that made me realize we are going to have a lot of ID casting soon since we need to switch between chalk and hir_def back and forth for database queries 😓 so thats gonna be fun